src/common/btree.c

```diff
@@ -1495,7 +1495,8 @@ btr_probe(struct btr_context *tcx, dbtree_probe_opc_t probe_opc,
 	case BTR_PROBE_GT:
 		if (cmp & BTR_CMP_GT) {
 			if ((intent == DAOS_INTENT_UPDATE ||
-			     intent == DAOS_INTENT_PUNCH) &&
+			     intent == DAOS_INTENT_PUNCH ||
+			     probe_opc & BTR_PROBE_MATCHED) &&
 			    !(cmp & BTR_CMP_MATCHED))
```

Comment:
is this right, or it should be like this?
 if ((intent == ...|| intent == ...) && (probe_opc & MATCHED) && !(cmp & BTR_CMP_MATCHED))

Reply:
It is the expected logic. The new added logic means that:
if the probe sponsor (mostly for fetch case) wants to locate a matched object, and if the found one does not match, then we do not need to check the target's availability.